pkg/rpcclient/nns/contract_test.go

```diff
@@ -195,3 +224,185 @@ func TestGetAllRecordsExpanded(t *testing.T) {
+				Values: []stackitem.Item{
+					stackitem.Make("n3"),
+					stackitem.Make("aaaaaa"),
+					stackitem.Make("cool"),
+				},
+			}),
+		},
+	}
+	iter, err = nns.Roots()
+	require.NoError(t, err)
+
+	ta.err = errors.New("")
+	err = iter.Terminate()
+	require.NoError(t, err)
+}
+
+func TestUpdate(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	nef := []byte{0x01, 0x02, 0x03}
+	manifest := "manifest data"
+
+	ta.err = errors.New("test error")
+	_, _, err := nns.Update(nef, manifest)
+	require.Error(t, err)
+
+	// Test successful update
+	ta.err = nil
+	ta.txh = util.Uint256{0x04, 0x05, 0x06}
+	txh, vub, err := nns.Update(nef, manifest)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestAddRoot(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	root := "example.root"
+	params, err := smartcontract.NewParameterFromValue(root)
+	require.NoError(t, err)
+	ta.err = errors.New("test error")
+	_, _, err = nns.AddRoot(params.Value.(string))
+	require.Error(t, err)
+
+	// Test success case
+	ta.err = nil
+	ta.txh = util.Uint256{0x07, 0x08, 0x09}
+	txh, vub, err := nns.AddRoot(root)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestSetPrice(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	priceList := []any{100, 200}
+	_, err := smartcontract.NewParameterFromValue(priceList)
+	require.NoError(t, err)
+	ta.err = errors.New("test error")
+	_, _, err = nns.SetPrice(priceList)
+	require.Error(t, err)
+
+	// Test success case
+	ta.err = nil
+	ta.txh = util.Uint256{0x0A, 0x0B, 0x0C}
+	txh, vub, err := nns.SetPrice(priceList)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestRegister(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	name := "example.neo"
+	owner := util.Uint160{0x0D, 0x0E, 0x0F}
+
+	ta.err = errors.New("test error")
+	_, _, err := nns.Register(name, owner)
+	require.Error(t, err)
+
+	// Test success case
+	ta.err = nil
+	ta.txh = util.Uint256{0x10, 0x11, 0x12}
+	txh, vub, err := nns.Register(name, owner)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestRenew(t *testing.T) {
+	ta := &testAct{}
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	name := "example.neo"
+
+	ta.err = errors.New("test error")
+	_, _, err := nns.Renew(name)
+	require.Error(t, err)
+
+	// Test success case
+	ta.err = nil
+	ta.txh = util.Uint256{0x13, 0x14, 0x15}
+	txh, vub, err := nns.Renew(name)
+	require.NoError(t, err)
+	require.Equal(t, ta.txh, txh)
+	require.Equal(t, ta.vub, vub)
+}
+
+func TestIntTransactions(t *testing.T) {
+	ta := new(testAct)
+	nns := New(ta, util.Uint160{1, 2, 3})
+
+	for _, fun := range []func(nef []byte, manifest string) (*transaction.Transaction, error){
+		nns.UpdateTransaction,
+		nns.UpdateUnsigned,
```

Comment:
Test name is `IntTransactions`, and the content is about update.

Reply:
And you already have the test for `update`, use it for `UpdateTransaction` and `UpdateUnsigned` tests.